### DISCUSSION TITLE
need some help to build.

### DISCUSSION THREAD

**Discussion by dotor1** [zhongfly/mpv-build#79] [Q&A]
Hello,

I'm having some trouble building mpv for win.

In Ubuntu, I have download the shinchiro git, and overwrite your mpv-winbuild on shinchiro's.

then I run build.sh , I have failed.

maybe something wrong, like SINGLE_SOURCE_LOCATION, RUSTUP_LOCATION,....

Do I have to download them before run build.sh?

mpv build for linux did well, but mpv build for windows is tricky.

If I have to install before run 'build.sh', would u let me know?

thx, bro

cf. I did the following before run 'build.sh'.
apt-get install build-essential checkinstall bison flex gettext git mercurial subversion ninja-build gyp cmake yasm nasm automake pkgconf libtool libtool-bin gcc-multilib g++-multilib clang lld libc++1 libc++abi1 libgmp-dev libmpfr-dev libmpc-dev libgcrypt-dev gperf ragel texinfo autopoint re2c asciidoc python3-pip docbook2x unzip p7zip-full curl

pip3 install rst2pdf meson mako jsonschema

**Comment by zhongfly** [comment]
You should provide the error message. I can only guess that this is caused by not having some package installed, such as `wget` `xz-utils`
After all need packages installed, run `build.sh -t '[target]' -c '[compiler]'`
`[target]` can be `32` `64` `64-v3` `all-64` `all`
`[compiler]` can be `clang` or `gcc`

**Reply by zhongfly** [reply]
To be honest, I'm only using arch linux to build, I don't know much about ubuntu, you'll need to solve the problem yourself based on the error messages.

**Comment by zhongfly** [comment]
In addition, please install the latest cmake, cmake in Ubuntu repositories may not be the latest version.
You can use the search engine to search for how to install the latest version of cmake

**Reply by dotor1** [reply]
thx for ur reply.

as u metioned, I have installed the latest version of cmake.

my error is something like this.

something to patch is wrong.

I don't know what is wrong precisely...

any advice?

![image](https://github.com/zhongfly/mpv-winbuild/assets/96975946/9e6ce1c9-45b4-4ea0-b467-9f63a4bf0037)

**Reply by zhongfly** [reply]
It has told you: run two commands
```
git config --global user.name "Your name"
git config --global user.email "Your email"
```
Input any name or email,no need to be real name or email
I think it says enough, it tells you exactly what you should do.

**Reply by dotor1** [reply]
thx 4 ur reply
